partiql-parser/src/main/kotlin/org/partiql/parser/internal/V1PartiQLParserDefault.kt

```diff
@@ -0,0 +1,2197 @@
+            graphMatchNode(prefilter, variable, label)
+        }
+
+        private fun visitRestrictor(ctx: GeneratedParser.PatternRestrictorContext?): GraphRestrictor? {
+            if (ctx == null) return null
+            return when (ctx.restrictor.text.lowercase()) {
+                "trail" -> GraphRestrictor.TRAIL()
+                "acyclic" -> GraphRestrictor.ACYCLIC()
+                "simple" -> GraphRestrictor.SIMPLE()
+                else -> throw error(ctx, "Unrecognized pattern restrictor")
+            }
+        }
+
+        /**
+         *
+         * TABLE REFERENCES & JOINS & FROM CLAUSE
+         *
+         */
+        override fun visitFromClause(ctx: GeneratedParser.FromClauseContext): From = translate(ctx) {
+            val tableRefs = visitOrEmpty<FromTableRef>(ctx.tableReference())
+            from(tableRefs)
+        }
+
+        override fun visitTableBaseRefSymbol(ctx: PartiQLParser.TableBaseRefSymbolContext): FromTableRef = translate(ctx) {
+            val expr = visitAs<Expr>(ctx.source)
+            val asAlias = visitSymbolPrimitive(ctx.symbolPrimitive())
+            fromExpr(expr, FromType.SCAN(), asAlias, null)
+        }
+
+        override fun visitTableBaseRefClauses(ctx: PartiQLParser.TableBaseRefClausesContext): FromTableRef = translate(ctx) {
+            val expr = visitAs<Expr>(ctx.source)
+            val asAlias = ctx.asIdent()?.let { visitSymbolPrimitive(it.symbolPrimitive()) }
+            val atAlias = ctx.atIdent()?.let { visitSymbolPrimitive(it.symbolPrimitive()) }
+            fromExpr(expr, FromType.SCAN(), asAlias, atAlias)
+        }
+
+        override fun visitTableBaseRefMatch(ctx: PartiQLParser.TableBaseRefMatchContext): FromTableRef = translate(ctx) {
+            val expr = visitAs<Expr>(ctx.source)
+            val asAlias = ctx.asIdent()?.let { visitSymbolPrimitive(it.symbolPrimitive()) }
+            val atAlias = ctx.atIdent()?.let { visitSymbolPrimitive(it.symbolPrimitive()) }
+            fromExpr(expr, FromType.SCAN(), asAlias, atAlias)
+        }
+
+        override fun visitTableUnpivot(ctx: PartiQLParser.TableUnpivotContext): FromTableRef = translate(ctx) {
+            val expr = visitAs<Expr>(ctx.expr())
+            val asAlias = ctx.asIdent()?.let { visitSymbolPrimitive(it.symbolPrimitive()) }
+            val atAlias = ctx.atIdent()?.let { visitSymbolPrimitive(it.symbolPrimitive()) }
+            fromExpr(expr, FromType.UNPIVOT(), asAlias, atAlias)
+        }
+
+        override fun visitTableWrapped(ctx: PartiQLParser.TableWrappedContext): FromTableRef = translate(ctx) {
+            visitAs<FromTableRef>(ctx.tableReference())
+        }
+
+        override fun visitTableCrossJoin(ctx: PartiQLParser.TableCrossJoinContext): FromTableRef = translate(ctx) {
+            val lhs = visitAs<FromTableRef>(ctx.lhs)
+            val rhs = visitAs<FromTableRef>(ctx.rhs)
+            // TODO support other CROSS JOIN variants (e.g. LEFT CROSS JOIN)
```

Comment:
The previous ANTLR parser visitor would parse `l LEFT CROSS JOIN r` to the AST equivalent of `l LEFT JOIN r ON TRUE`. This was mentioned in the PartiQL spec in section 5.4 (https://partiql.org/dql/joins.html#left-join), but I couldn't find it mentioned in the SQL1999 spec. It's not yet clear to me what other JOIN types (e.g. `RIGHT`, `LEFT OUTER`, `INNER`, `FULL`, etc.) should support this behavior so I've left it out of the new ANTLR parser visitor for now.

Reply:
From offline discussion, will change the ANTLR parsing rules slightly:
1. Remove the optional `joinType` before the `CROSS JOIN` since that's what is defined in SQL99 spec
2. Add an explicit rule for `LEFT CROSS JOIN` since that's what is defined in the PartiQL spec
3. Add TODO linking Vladimir's issue for the other cross join variant investigation -- https://github.com/partiql/partiql-lang-kotlin/issues/1013